pkg/controller/pxc/upgrade.go

```diff
@@ -240,7 +240,24 @@ func (r *ReconcilePerconaXtraDBCluster) smartUpdate(sfs api.StatefulApp, cr *api
 		return errors.Wrap(err, "failed to get current sfs")
 	}
 
-	if currentSet.Status.UpdatedReplicas >= currentSet.Status.Replicas {
+	list := corev1.PodList{}
+	if err := r.client.List(context.TODO(),
+		&list,
+		&client.ListOptions{
+			Namespace:     currentSet.Namespace,
+			LabelSelector: labels.SelectorFromSet(sfs.Labels()),
+		},
+	); err != nil {
+		return errors.Wrap(err, "get pod list")
+	}
+	update := false
```

Comment:
can we please rename var? for better code reading
```suggestion
	updateInProgress := false
```